akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -268,6 +268,62 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
+  private val buf = {
+    val b = Vector.newBuilder[T]
+    b.sizeHint(n)
+    b
+  }
+  var bufferedElements = 0
+
+  override def onPush(elem: T, ctx: Context[immutable.Seq[T]]): SyncDirective = {
+    buf += elem
+    bufferedElements += 1
+    if (bufferedElements < n) {
+      ctx.pull()
+    } else if (bufferedElements == n) {
+      ctx.push(buf.result())
+    } else {
```

Comment:
`bufferedElements > n` does happen because in the branch `bufferedElements == n` we don't clear the buffer. This is because the buffer is needed to be kept in case `step > n`.